connection.go

```diff
@@ -46,6 +45,7 @@ func (c *connection) doTunnelClose(err error) {
 		return
 	}
 
+	metrics.IncSMTotalRemoveConnectionsForWS(c.session.clientKey, c.addr.Network(), c.addr.String())
```

Comment:
See https://github.com/rancher/remotedialer/pull/74/files/5e8dc8524011d90cf1cdcbd17713cb57f2d931a5#r1553397664 regarding this change

Reply:
Won't this result in a double increment in `pipe`, which is currently the only other place calling `doTunnelClose()` directly?

Reply:
Understood: the caller does that now.

Reply:
IncSMTotalRemoveConnectionsForWS

Reply:
FTR: before these changes, `doTunnelClose` is being called 3 times in total (2 from `pipe`, 1 from `clientDial` by deferring `conn.Close`, which then calls `conn.tunnelClose`). Of those 3 calls, only one of them will be effective, since there is a protection (checking `c.err != nil`) for the function being called before. Since I placed the `metrics` recording after this protection, it will keep being increased only in one of those calls.